velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -485,6 +491,85 @@ TEST_F(DateTimeFunctionsTest, yearTimestampWithTimezone) {
           "year(c0)", std::nullopt, std::nullopt));
 }
 
+TEST_F(DateTimeFunctionsTest, weekDate) {
+  const auto weekDate = [&](const StringView& date) {
+    auto week =
+        evaluateOnce<int64_t>("week(c0)", std::make_optional(parseDate(date)))
+            .value();
+    auto weekOfYear =
+        evaluateOnce<int64_t>(
+            "week_of_year(c0)", std::make_optional(parseDate(date)))
+            .value();
+    VELOX_CHECK(
+        week == weekOfYear, "week and week_of_year must return the same value");
+    return week;
+  };
+
+  EXPECT_EQ(1, weekDate("1919-12-31"));
+  EXPECT_EQ(1, weekDate("1920-01-01"));
+  EXPECT_EQ(1, weekDate("1920-01-04"));
+  EXPECT_EQ(2, weekDate("1920-01-05"));
+  EXPECT_EQ(53, weekDate("1960-01-01"));
+  EXPECT_EQ(53, weekDate("1960-01-03"));
+  EXPECT_EQ(1, weekDate("1960-01-04"));
+  EXPECT_EQ(1, weekDate("1969-12-31"));
+  EXPECT_EQ(1, weekDate("1970-01-01"));
+  EXPECT_EQ(1, weekDate("0001-01-01"));
+  EXPECT_EQ(52, weekDate("9999-12-31"));
+}
+
+TEST_F(DateTimeFunctionsTest, week) {
+  const auto weekTimestamp = [&](const StringView& time) {
+    auto timestampInSeconds = util::fromTimeString(time) / 1'000'000;
+    auto week =
+        evaluateOnce<int64_t>(
+            "week(c0)",
+            std::make_optional(Timestamp(timestampInSeconds * 100'000'000, 0)))
+            .value();
+    auto weekOfYear =
+        evaluateOnce<int64_t>(
+            "week_of_year(c0)",
+            std::make_optional(Timestamp(timestampInSeconds * 100'000'000, 0)))
+            .value();
+    VELOX_CHECK(
+        week == weekOfYear, "week and week_of_year must return the same value");
+    return week;
+  };
+
+  EXPECT_EQ(1, weekTimestamp("00:00:00"));
+  EXPECT_EQ(10, weekTimestamp("11:59:59"));
+  EXPECT_EQ(51, weekTimestamp("06:01:01"));
+  EXPECT_EQ(24, weekTimestamp("06:59:59"));
+  EXPECT_EQ(27, weekTimestamp("12:00:01"));
+  EXPECT_EQ(7, weekTimestamp("12:59:59"));
+}
+
+TEST_F(DateTimeFunctionsTest, weekTimestampWithTimezone) {
+  const auto weekTimestampTimezone = [&](const StringView& time,
```

Comment:
StringView's can be passed by value, but here it is confusing to use Velox's StringView. Consider changing to const char* or std::string_view.